packages/hagrid/hagrid/cli.py

```diff
@@ -351,40 +353,164 @@ def launch(args: TypeTuple[str], **kwargs: Any) -> None:
+    task = progress_bar.tasks[0]
+    if "Error response from daemon: " in line:
+        if progress_bar:
+            progress_bar.update(
+                0,
+                description=f"❌ [bold red]{cmd_name}[/bold red] [{task.completed} / {task.total}]",
+                refresh=True,
+            )
+            progress_bar.update(0, visible=False)
+            progress_bar.console.clear_live()
+            progress_bar.console.quiet = True
+            progress_bar.stop()
+            console = rich.get_console()
+            progress_bar.console.quiet = False
+            console.print(f"\n\n [red] ERROR [/red]: [bold]{line}[/bold]\n")
+        os.killpg(os.getpgid(pid), sys_signal.SIGTERM)
+        raise Exception
+
+
+def check_pulling(line: str, cmd_name: str, progress_bar: Progress) -> None:
+    task = progress_bar.tasks[0]
+    if "Pulling" in line and "fs layer" not in line:
+        progress_bar.update(
+            0,
+            description=f"⌛ [bold]{cmd_name} [{task.completed} / {task.total+1}]",
+            total=task.total + 1,
+            refresh=True,
+        )
+    if "Pulled" in line:
+        progress_bar.update(
+            0,
+            description=f"⌛ [bold]{cmd_name} [{task.completed + 1} / {task.total}]",
+            completed=task.completed + 1,
+            refresh=True,
+        )
+        if progress_bar.finished:
+            progress_bar.update(
+                0,
+                description=f"✅ [bold green]{cmd_name} [{task.completed} / {task.total}]",
+                refresh=True,
+            )
 
-    Args:
-        cmds (list): list of commands to be executed
-        dry_run (bool, optional): If `True` only displays cmds to be executed. Defaults to False.
-    """
-    process_list: TypeList = []
-    console = rich.get_console()
 
-    username, password = (
-        extract_username_and_pass(cmds[0]) if len(cmds) > 0 else ("-", "-")
-    )
-    # display VM credentials
-    console.print(generate_user_table(username=username, password=password))
+def check_building(line: str, cmd_name: str, progress_bar: Progress) -> None:
+    current_build_steps = 55
+    pattern = re.search("^#[0-9]{1,3}", line)
+    if pattern:
+        current_step = int(pattern.group()[1:])
+        task = progress_bar.tasks[0]
+        if current_step > task.completed:
+            progress_bar.update(
+                0,
+                description=f"⌛ [bold]{cmd_name} [{current_step} / {current_build_steps}]",
+                completed=current_step,
+                total=current_build_steps,
+                refresh=True,
+            )
+            if progress_bar.finished:
+                progress_bar.update(
+                    0,
+                    description=f"✅ [bold green]{cmd_name} [{task.completed} / {task.total}]",
+                    refresh=True,
+                )
+
+
+def check_launching(line: str, cmd_name: str, progress_bar: Progress) -> None:
```

Comment:
We need to improve how to track docker logs in the launching step since the output in stdout/stderr seems to be different from **--tail=True**/**--tail=False**.

Reply:
Also, I think we can remove the --silent flag altogether, and just use the --tail flag to show logs if its set to true and just show the progress of all steps if --tail is false.

Reply:
Sure!

Reply:
[Makes silent mode only available while tail=false](https://github.com/OpenMined/PySyft/pull/6900/commits/c6e4c05082dd15b403af83f9952945deada57a72)
I didn't remove the `silent` flag entirely since it's used by different methods outside of the scope of docker deployment.
However, now we force it to be True only when tail=False for docker deployment.